### ISSUE TITLE
[otbn] ISA orthogonality: BN.MOV vs. alternatives

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#6482]
The OTBN bignum ISA subset has currently at least two ways of copying data between wide registers (WDRs) with direct addressing:

* `BN.MOV <wrd>, <wrs>`
* `BN.ADDI <wrd>, <wrs>, 0`

With two differences, however:
* BN.MOV will preserve integrity protection (the whole data word + integrity protection bits are copied untouched).
* BN.ADDI will strip off integrity protection (since the data passes through the ALU), but will set flags instead.

Generally, having two instructions doing almost the same thing is something we want to avoid in the ISA design, until there are good reasons for having both. 

@felixmiller, since you know the assembly code best: Can we make `BN.MOV` (and then also `BN.MOVR`) have a flag group argument and set flags, making it an alias of `BN.ADDI`?

Just for completeness: We can special-case `BN.ADDI <wrd>, <wrs>, 0` in the implementation to retain integrity protection (and we will have to do that once we make BN.MOV an alias of it). But it would be even better if we can avoid a special case and instead do a simplification.

> estimate 2

**Comment by rswarbrick** [comment]
Other options:

- Tooling: Spit out a warning from the assembler on `BN.ADDI _, _, 0` with a way to waive it.
- Improve documentation (probably not enough?)
- Actually disallow `BN.ADDI _, _, 0` (raise a bad instruction error) to force programmers to use `BN.MOV`. If we do this, we need to make sure no-one is ever going to need to set the flags like this (maybe you could use e.g. `BN.AND` for a similar purpose if necessary)

**Comment by imphil** [comment]
Felix, Vladimir and I discussed this today and we came to the conclusion that we want to keep the ISA as-is. Having BN.MOV(R) available to transfer data between registers if there's register pressure without any side effects (e.g. setting flags) is helpful and commonly used in existing code.

We should:
* Recommend the use of BN.MOV over BN.ADDI for register moves in the coding guidelines, to be created in https://github.com/lowRISC/opentitan/issues/3314.
* Optionally, issue a warning in the assembler or in to-be-developed lint tooling.
